Modules/readline.c

```diff
@@ -576,6 +576,7 @@ readline_set_completer_delims(PyObject *module, PyObject *string)
     if (break_chars) {
         free(completer_word_break_characters);
         completer_word_break_characters = break_chars;
+        rl_basic_word_break_characters = break_chars;
```

Comment:
Okay so it's kinds of implementation detail.
Since we enable the flag for libedit in following way.
https://github.com/python/cpython/blob/2e632fa07d13a58be62f59be4e656ad58b378f9b/configure.ac#L5916

What about changing the code for the libedit implementation only?
```suggestion
#ifdef WITH_EDITLINE
         // Comment why this is needed
        rl_basic_word_break_characters = break_chars;
#endif
```

Reply:
From the comments in the file:

```
 * This emulation library is not 100% API compatible with the "real" readline
 * and cannot be detected at compile-time,
 * hence we use a runtime check to detect if the Python readline module is
 * linked to libedit.
```

It seems like we are not able to detect if `libedit` is used at compile time. I doubt if the macro protection would actually work in all cases.

Reply:
Do our official binaries build with libedit ever?

Reply:
> It seems like we are not able to detect if libedit is used at compile time

I am not sure when the comment is added.
But we can detect editline at build-time from Python 3.10
https://docs.python.org/3/using/configure.html?highlight=with_editline#cmdoption-with-readline

Reply:
https://github.com/python/cpython/pull/24189

Reply:
> Do our official binaries build with libedit ever?

Even if our official build does not use it, we still consider downstream distros :)

Reply:
That's actually not what I meant. What I meant is - we could have been using `libedit` without `WITH_EDITLINE`. So even if we do not define `WITH_EDITLINE`, we could still be using `libedit`. I just confirmed on my mac that this change will break the test.

The reason I quoted that comment was that I believed such case exists - where we don't know if `libedit` is being used at compile time.

Reply:
Got it, then should we consider using both `using_libedit_emulation` and `WITH_EDITLINE`.
If you think that we don't have to consider side effect from the implementation detail, then we can just add a comment why we do this.

Reply:
```suggestion
#ifdef WITH_EDITLINE
        rl_basic_word_break_characters = break_chars;
#else
       if (using_libedit_emulation) rl_basic_word_break_characters = completer_word_break_characters 
#endif
```

Reply:
Sorry I did not use your suggestion directly. I do like the brackets for the if statement. I applied the protection to avoid side effects for GNU readline on both cases.